tests/examples/tokens/test_erc1155ownable_ETHtester.py

```diff
@@ -0,0 +1,316 @@
+
+    assert (lambda: NFT_contract.balanceOfBatch([w3.eth.accounts[1],w3.eth.accounts[1],w3.eth.accounts[1]],[11,12,13], transact={"from": w3.eth.accounts[1]}) == [0,0,1])
+
+    # burn again, should revert
+    assert_tx_failed(lambda: NFT_contract.burnBatch([1,2],[1,1],transact={"from": w3.eth.accounts[1]}))
+
+    assert (lambda: NFT_contract.balanceOfBatch([w3.eth.accounts[1],w3.eth.accounts[1],w3.eth.accounts[1]],[1,2,3], transact={"from": w3.eth.accounts[1]}) == [0,0,1])
+
+
+def test_approval_functions(NFT_contract, w3, test_mint_batch, assert_tx_failed):
+
+    # self-approval by the owner
+    assert_tx_failed(lambda: NFT_contract.setApprovalForAll(w3.eth.accounts[5],w3.eth.accounts[5],True, transact={"from": w3.eth.accounts[5]}))
+
+    # let's approve and operator for somebody else's account
+    assert_tx_failed(lambda: NFT_contract.setApprovalForAll(w3.eth.accounts[0],w3.eth.accounts[5],True, transact={"from": w3.eth.accounts[3]}))
+
+    # set approval correctly
+    NFT_contract.setApprovalForAll(w3.eth.accounts[0],w3.eth.accounts[5],True)
+
+    # check approval
+    NFT_contract.isApprovedForAll(w3.eth.accounts[0],w3.eth.accounts[5])
+
+    # remove approval
+    NFT_contract.setApprovalForAll(w3.eth.accounts[0],w3.eth.accounts[5],False)
+
+
+def test_max_batch_size_violation(NFT_contract, w3, assert_tx_failed):
+    TOTAL_BAD_BATCH = 200
+    ids = []
+    amounts = []
+    for i in range(1,TOTAL_BAD_BATCH):
+        ids.append(i)
+        amounts.append(1)
+
+    assert_tx_failed(lambda: NFT_contract.mintBatch(w3.eth.accounts[1], ids, amounts, '', transact={"from": w3.eth.accounts[0]}))
+
+
+# Transferring back and forth
+
+def test_ownership_functions(NFT_contract, w3, assert_tx_failed,tester):
+    owner, a1, a2 = w3.eth.accounts[0:3]
+    print(owner,a1,a2)
+    print('___owner___', NFT_contract.owner())
+    # change owner from account 0 to account 1 and back
+    
+    assert NFT_contract.owner() == owner
+    assert_tx_failed(lambda: NFT_contract.transferOwnership(w3.eth.accounts[1], transact={"from": a2}))
+    
+    # try to transfer ownership to current owner
+    assert_tx_failed(lambda: NFT_contract.transferOwnership(w3.eth.accounts[0]))
+    # try to transfer ownership to ZERO ADDRESS
+    assert_tx_failed(lambda: NFT_contract.transferOwnership("0x0000000000000000000000000000000000000000"))
+
+    # Transfer ownership to account 1    
+    NFT_contract.transferOwnership(w3.eth.accounts[1])
+    assert (lambda: NFT_contract.owner() == w3.eth.accounts[1])
+
+def test_renounce_ownership(NFT_contract, w3, assert_tx_failed):
+    
+    assert NFT_contract.owner() == w3.eth.accounts[0]
+    # try to transfer ownership from non-owner account
+    assert_tx_failed(lambda: NFT_contract.renounceOwnership(transact={"from": w3.eth.accounts[2]}))
+
+    NFT_contract.renounceOwnership(transact={"from": w3.eth.accounts[0]})
+
+    assert NFT_contract.owner() == None
```

Comment:
This line should fail

Reply:
After renounce ownership there should be no owner I believe.
Not entirely sure what you mean. Should line 316 fail because we expect ZERO_ADDRESS?

Reply:
Fixed and tested!